runtime/pangolin-parachain/src/bridge_messages/pangolin.rs

```diff
@@ -0,0 +1,233 @@
+// --- crates.io ---
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+// --- paritytech ---
+use frame_support::{
+	weights::{DispatchClass, Weight},
+	RuntimeDebug,
+};
+use sp_runtime::{traits::Zero, FixedPointNumber, FixedU128};
+use sp_std::ops::RangeInclusive;
+// --- darwinia-network ---
+use crate::*;
+use bp_messages::{source_chain::TargetHeaderChain, target_chain::*, *};
+use bp_runtime::{Chain, ChainId, PANGOLIN_CHAIN_ID, PANGOLIN_PARACHAIN_CHAIN_ID};
+use bridge_runtime_common::messages::{self, source::*, target::*, *};
+use pallet_bridge_messages::EXPECTED_DEFAULT_MESSAGE_LENGTH;
+
+/// Message verifier for PangolinParachain -> Pangolin messages.
+pub type ToPangolinMessageVerifier = FromThisChainMessageVerifier<WithPangolinMessageBridge>;
```

Comment:
Compare with https://github.com/darwinia-network/darwinia-common/blob/main/node/runtime/pangoro/src/pangolin_messages.rs#L67. Loss generic param R, is that expected?